liquibase-couchbase/src/test/java/liquibase/ext/couchbase/operator/ChangeLogOperatorTest.java

```diff
@@ -0,0 +1,216 @@
+        RanChangeSet ranChangeSet2 = createChangeset(changeLog2);
+        expectedChangesets.add(ranChangeSet1);
+        expectedChangesets.add(ranChangeSet2);
+
+        when(serviceProvider.getScopeOfCollection(CHANGE_LOG_COLLECTION)).thenReturn(testScope);
+        when(testScope.query(eq(SELECT_ALL_CHANGELOGS_N1QL), any(QueryOptions.class))).thenReturn(queryResult);
+        when(queryResult.rowsAs(CouchbaseChangeLog.class)).thenReturn(Arrays.asList(changeLog1, changeLog2));
+
+        assertEquals(expectedChangesets, changeLogOperator.getAllChangeLogs());
+    }
+
+    private static CouchbaseChangeLog createChangeLog(String fileName, Context context) {
+        return CouchbaseChangeLog.builder()
+                .fileName(fileName)
+                .id("id")
+                .author("author")
+                .dateExecuted("2023.06.02 24:59:59")
+                .context(context)
+                .labels(emptySet())
+                .build();
+    }
+
+    @NonNull
+    private static RanChangeSet createChangeset(CouchbaseChangeLog changeLog) {
+        Date date = new Date(2023, Calendar.JULY, 2, 23, 59, 59);
+        return new RanChangeSet(
+                changeLog.getFileName(),
+                changeLog.getId(),
+                changeLog.getAuthor(),
+                null,
+                date,
+                null,
+                null,
+                null,
+                null,
+                new ContextExpression(changeLog.getContext().getOriginalString()),
+                new Labels(changeLog.getLabels()),
+                null
+        );
+    }
+
+    @Test
+    void Should_remove_changeset_from_history() {
+        when(changeSet.getFilePath()).thenReturn("file");
+        when(changeSet.getId()).thenReturn("id");
+        when(changeSet.getAuthor()).thenReturn("author");
+        when(serviceProvider.getServiceCollection(CHANGE_LOG_COLLECTION)).thenReturn(collection);
+
+        changeLogOperator.removeChangeSetFromHistory(changeSet);
+
+        verify(collection, times(1)).remove("file::id::author");
+    }
+
+    @Test
+    void Should_tag_last_changeset() {
+        String changesetId = "id1";
+        String tag = "version-1.0";
+        JsonObject jsonObject = JsonObject.create().put("id", changesetId);
+
+        when(serviceProvider.getScopeOfCollection(CHANGE_LOG_COLLECTION)).thenReturn(testScope);
+        when(serviceProvider.getServiceCollection(CHANGE_LOG_COLLECTION)).thenReturn(collection);
+        when(testScope.query(eq(SELECT_LAST_CHANGE_SET_ID_N1QL), any(QueryOptions.class))).thenReturn(queryResult);
+        when(queryResult.rowsAsObject()).thenReturn(singletonList(jsonObject));
+
+        changeLogOperator.tagLastChangeSet(tag);
+
+        verify(collection, times(1)).mutateIn(eq(changesetId), any(List.class));
+    }
+
+    @Test
+    void Should_create_changeLogOperator_when_databaseProvided() {
+        new ChangeLogOperator(database);
```

Comment:
Missing an assertion here, don't we?

Reply:
Actually this was made to test constructor. But I think we can remove constructore coverage from jacoco reports. What do you think?

Reply:
Actually all code smells in my request were all about "remove  public keyword", except this one with no assertion.

Reply:
Done